docs/deployment/performance.md

```diff
@@ -0,0 +1,584 @@
+</ContentPage>
+```
+
+This is wasteful and the [`VerticalStackLayout`](xref:Microsoft.Maui.Controls.VerticalStackLayout) element should be removed, as shown in the following example:
+
+```xaml
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="MyMauiApp.MainPage">
+    <Image Source="waterfront.jpg" />
+</ContentPage>
+
+```
+
+In addition, don't attempt to reproduce the appearance of a specific layout by using combinations of other layouts, as this results in unnecessary layout calculations being performed. For example, don't attempt to reproduce a [`Grid`](xref:Microsoft.Maui.Controls.Grid) layout by using a combination of [`HorizontalStackLayout`](xref:Microsoft.Maui.Controls.HorizontalStackLayout) elements. The following example shows an example of this bad practice:
+
+```xaml
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="MyMauiApp.MainPage">
+    <VerticalStackLayout>
+        <HorizontalStackLayout>
+            <Label Text="Name:" />
+            <Entry Placeholder="Enter your name" />
+        </HorizontalStackLayout>
+        <HorizontalStackLayout>
+            <Label Text="Age:" />
+            <Entry Placeholder="Enter your age" />
+        </HorizontalStackLayout>
+        <HorizontalStackLayout>
+            <Label Text="Occupation:" />
+            <Entry Placeholder="Enter your occupation" />
+        </HorizontalStackLayout>
+        <HorizontalStackLayout>
+            <Label Text="Address:" />
+            <Entry Placeholder="Enter your address" />
+        </HorizontalStackLayout>
+    </VerticalStackLayout>
+</ContentPage>
+
+```
+
+This is wasteful because unnecessary layout calculations are performed. Instead, the desired layout can be better achieved using a [`Grid`](xref:Microsoft.Maui.Controls.Grid), as shown in the following example:
+
+```xaml
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="MyMauiApp.MainPage">
+    <Grid ColumnDefinitions="100,*"
+          RowDefinitions="30,30,30,30">
+        <Label Text="Name:" />
+        <Entry Grid.Column="1"
+               Placeholder="Enter your name" />
+        <Label Grid.Row="1"
+               Text="Age:" />
+        <Entry Grid.Row="1"
+               Grid.Column="1"
+               Placeholder="Enter your age" />
+        <Label Grid.Row="2"
+               Text="Occupation:" />
+        <Entry Grid.Row="2"
+               Grid.Column="1"
+               Placeholder="Enter your occupation" />
+        <Label Grid.Row="3"
+               Text="Address:" />
+        <Entry Grid.Row="3"
+               Grid.Column="1"
+               Placeholder="Enter your address" />
+    </Grid>
+</ContentPage>
+```
+
+## Optimize layout performance
```

Comment:
@hartez Is this section accurate/worth keeping?

Reply:
Line 122 isn't always true, and can probably be dropped.